contrib/git.lua

```diff
@@ -0,0 +1,245 @@
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+]]
+
+--[[
+   Installation
+     1. Copy this file into ~/.config/darktable/lua/
+     2. Add `require "git"` to a new line in ~/.config/darktable/luarc
+     3. Restart darktable
+     4. Run your mouse cursor over images in the lighttable view - color tags will update.
+     5. Use the "VCS/Git" gui box on the right to add/rm files to/from a Git repo.
+     6. Lower left "Image Information" box will have two new rows for Git file status.
+]]
+
+--[[
+   I use Git for the archival, backup, and distribution of my image
+   files and my XMP sidecar files.
+
+   I had two needs from the DT interface:
+     1. At a glance - know the status of an image file.
+        a. If it is not tracked by a Git repo - red/danger - no backup or version
+           control is in place.
+        b. If it is tracked by a Git repo but not committed or is modified - yellow/warning.
+        c. If it is tracked, and is committed - green/good.
+     2. A simple interface to add/remove the image files to/from the relevant repo.
+
+   Notes:
+     1. This is not a full-feature Git GUI. There is no commit/revert/push/pull/etc.
+        You'll need to find/perform that elsewhere.
+     2. There is a "hole" in that an image may be committed to the local repo, and not
+        pushed to the remote. This interface does not warn about such a condition.
+
+   What you get:
+     1. Color flags Red/Yellow/Green:  On image/folder import, and on lighttable view
+        mouseover, the Git status of an image file is queried and the color flags
+        are updated according to this logic:
+
+        a. Red    - local file not known to a Git repo
+        b. Yellow - staged file, or modified file
+        c. Green  - fully committed file
+
+        Note that the color flags say nothing about the status of the sidecar file.
+
+     2. Image Information box - In the "Image Information" box (bottom left of the
+        DT GUI) two new info rows are added:
+
+        a. "Git/image"   - An English string of the Git status of the image file
+        b. "Git/sidecar" - An English string of the Git status of the sidecar file
+
+     3. "VCS/Git" UI box - on the right side of the lighttable UI is a new box. This
+        allows for a Git "add" or "rm" of images currently selected.
+
+   TODO:
+     1. Are there native Lua Git bindings so that io.popen() and os.execute() can be avoided?
+]]
+
+dt       = require "darktable"
+dt_debug = require "darktable.debug" 
+
+-- for a given file, determine the git status, return a descriptive string
+local function git_status_of_file(path, filename)
+  local result
+  local cmd
+  local f
+  local output
+
+--	dt.print_log(path .. " / " .. filename)
+  result = os.execute("git -C " .. path .. " rev-parse --git-dir 2>&1 > /dev/null 2>&1 ")
+  if result == nil then
+    return "No Git repo"
```

Comment:
You should make your strings translatable, we have an example how to do that here:
https://github.com/darktable-org/lua-scripts/blob/master/examples/gettextExample.lua